### ISSUE TITLE
[pattern] overflow content: update truncation section

### ISSUE THREAD

**Issue by aagonzales** [carbon-design-system/carbon-website#2373]
Need to update truncation section to include all the helpful guidelines and tips from @dakahn lunch and learn session.
https://www.carbondesignsystem.com/patterns/overflow-content/#truncation

**Comment by dakahn** [comment]
From my notes

Generally truncation is considered a design-smell and not necessarily an outright accessibility violation. The technique's use needs to be evaluated for usability/accessibility on a case by case basis. There are no strictly defined rules around its use and behavior specifically.

That said, links, labels or descriptions on controls should never truncate. This is a clear violation of rule 2.4.4 since the link's destination is potentially ambiguous.

From the UX side of things many single lined truncated items on a page can create a frustrating experience potentially turning a 1 second task into a 30 second as the user is forced to focus or mouse over each one for a tooltip so they can make a decision/understand. When dealing with text 80 pixels wide truncated to fit a 65 pixel width container, you dont necessarily have an accessibility violation (unless its a link), but you have a big usability problem. Wrapping is preferred in these situations.

Super happy to jump on a call and help whoever authors this figure out and details or extend anything that's too vague 👍🏽
